### DISCUSSION TITLE
3.4.8 Warning regarding node_modules folder detecting something I don't understand

### DISCUSSION THREAD

**Discussion by SolidAnonDev** [tailwindlabs/tailwindcss#14145] [Help] [answered]
According to this PR: #14140 when broad glob patterns are used it makes some effort to determine if there are node_modules folders in the specified directory according to these criteria:

```
    We detect ** in the glob pattern
    and you didn't explicitly use node_modules in the glob pattern
    and we found files that include node_modules in the file path
    and no other globs exist that explicitly match the found file
```

However, I am using this in a Next.JS project to specify the `./src` folder using the glob pattern:

`./src/**/*.{ts,tsx}`

The build now throws the following warning:

```
Your 'content' configuration includes a pattern which looks like it's accidentally matching all of 'vendor' and can cause serious performance issues

Pattern: './src/**/*.ts'
```

I am confused by this because my node_modules folder is on the same level as `src` and thus, there are no node_modules folders in the src folder that I am aware of. 

My folder structure is something like

```
node_modules/
src/
├─ app/

```
FInally, I don't know what 'vendor' is or how I would diagnose that to begin with. I have no files in the `src` folder named vendor, just a bunch of `.tsx` files. It doesn't matter how I alter the config, or if I choose to only include `.tsx` files or use `.{ts,tsx}`, it throws the same warning in either case.

I kind of wanted to open an issue on this, but figured I would ask here first to see if anyone could figure out why this would throw this warning off the top of their heads.

**Comment by wongjn** [comment]
The message mentions a `vendor` folder, not `node_modules` – perhaps you have a `vendor` folder in your `src` folder somewhere?

**Comment by adamwathan** [comment, marked as answer]
This is a false positive in your case, `vendor` is the `node_modules` equivalent in PHP projects but we should probably find a way to not warn if it doesn’t actually look like a real PHP vendor directory.

**Reply by SolidAnonDev** [reply]
Interesting - I will have to check tomorrow but as far as I know I don't have a `vendor` folder in my `src` directory - is there something else that could be triggering this warning?

In any event that's good to know. Should I open an issue regarding this or leave it?

**Reply by adamwathan** [reply]
Ah if there’s not even a vendor folder there then definitely open an issue with a reproduction if you don’t mind 🙏

**Reply by SolidAnonDev** [reply]
In attempting a reproduction, I have found that any file or folder containing the word `vendor` will cause this warning.

In my case, this is due to the fact that I have a database entity called `vendor`, thus I have a few files and folders containing the word `vendor` like `vendor-service.ts` or `edit-vendors/` and so on

It wasn't until some playing around that I figured out that anything containing the word will trigger this warning. I have opened issue #14149 with this reproduction.

**Comment by philipp-spiess** [comment]
Closing this since should be fixed by https://github.com/tailwindlabs/tailwindcss/pull/14147 released in [v3.4.9](https://github.com/tailwindlabs/tailwindcss/releases/tag/v3.4.9).

Thanks for the report!!
